hooks/useMetaMaskBalances.ts

```diff
@@ -0,0 +1,37 @@
+import { useMetaMaskWallet } from "@/providers/MetaMaskWalletProvider";
+import { BridgedEthereumToken, EthereumToken } from "@/types";
+import { Balance, fetchMetaMaskBalance } from "@/utils";
+import { useEffect, useMemo, useState } from "react";
+
+export default function useMetaMaskBalances(
+	token1: EthereumToken | BridgedEthereumToken,
+	token2?: EthereumToken | BridgedEthereumToken
+): [Balance, Balance, () => void] {
+	const { wallet, selectedAccount } = useMetaMaskWallet();
+	const [balance1, setBalance1] = useState<Balance>(null);
+	const [balance2, setBalance2] = useState<Balance>(null);
+
+	const updateBalances = useMemo(() => {
+		if (!wallet || !selectedAccount?.address) return;
+
+		return async () => {
+			setBalance1(null);
+			fetchMetaMaskBalance(wallet, selectedAccount.address, token1).then(
+				setBalance1
+			);
+
+			if (token2) {
+				setBalance2(null);
+				fetchMetaMaskBalance(wallet, selectedAccount.address, token2).then(
+					setBalance2
+				);
+			}
```

Comment:
what's the use case for this?

Reply:
nice spotting, there is not really a use case for it atm, might come in handy in the future if we decide to go full-on MetaMask, plus I want to keep it similar to `useCENNZBalances`